src/components/common/CountDownTimer.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <div />
+</template>
+
+<script>
+import VueTimers from 'vue-timers/mixin';
+
+export default {
+  name: 'CountDownTimer',
+
+  props: {
+    timeout: {
+      type: Number,
+      default: 30000,
+    },
+    isLocked: {
+      type: Boolean,
+      default: false,
+    },
+    counter: {
+      type: Number,
+      default: 0,
+    },
+  },
+
+  computed: {
+    initialCounter() {
+      return this.timeout / 1000;
+    },
+
+    isRequestTimerRunning() {
+      return this.timers.request.isRunning;
+    },
+  },
+
+  watch: {
+    isLocked(newValue) {
+      if (!newValue) {
+        if (this.isRequestTimerRunning) {
```

Comment:
maybe move  `if (this.isRequestTimerRunning) {` checking to the beginning of `stopRequestTimer` method?

Reply:
if we have `if in if` it's a good moment for thinking :slightly_smiling_face: